Mathlib/RingTheory/Polynomial/Wronskian.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright (c) 2024 Jineon Back and Seewoo Lee. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jineon Baek, Seewoo Lee
+-/
+import Mathlib.Algebra.Polynomial.Derivative
+
+/-!
+# Wronskian of a pair of polynomial
+
+This file defines Wronskian of a pair of polynomial, which is `W(a, b) = ab' - a'b`.
+We also prove basic properties of it.
+
+## Main declarations
+
+- `wronskian_eq_sum_zero`: We have `W(a, b) = W(b, c)` when `a + b + c = 0`.
+- `wronskian.degree_lt_add`: Degree of Wronskian `W(a, b)` is strictly smaller than
+  the sum of degrees of `a` and `b`
+- `wronskian.natDegree_lt_add`: `natDegree` version of the above theorem. We need to assume that
+  the Wronskian is nonzero. (Otherwise, `a = b = 1` gives a counterexample.)
+-/
+
+noncomputable section
+
+open scoped Polynomial Classical
+
+open Polynomial
+
+variable {R : Type _} [CommRing R]
+
+/-- Wronskian: W(a, b) = ab' - a'b. -/
+def wronskian (a b : R[X]) : R[X] :=
+  a * (derivative b) - (derivative a) * b
+
+@[simp]
+theorem wronskian_zero_left (a : R[X]) : wronskian 0 a = 0 := by
+  simp_rw [wronskian]; simp only [MulZeroClass.zero_mul, derivative_zero, sub_self]
+
+@[simp]
+theorem wronskian_zero_right (a : R[X]) : wronskian a 0 = 0 := by
+  simp_rw [wronskian]; simp only [derivative_zero, MulZeroClass.mul_zero, sub_self]
+
+theorem wronskian_neg_left (a b : R[X]) : wronskian (-a) b = -wronskian a b := by
+  simp_rw [wronskian, derivative_neg]; ring
+
+theorem wronskian_neg_right (a b : R[X]) : wronskian a (-b) = -wronskian a b := by
+  simp_rw [wronskian, derivative_neg]; ring
+
+theorem wronskian_add_right (a b c : R[X]) : wronskian a (b + c) = wronskian a b + wronskian a c :=
+  by simp_rw [wronskian, derivative_add]; ring
+
+theorem wronskian_self (a : R[X]) : wronskian a a = 0 := by rw [wronskian, mul_comm, sub_self]
+
+theorem wronskian_anticomm (a b : R[X]) : wronskian a b = -wronskian b a := by
+  rw [wronskian, wronskian]; ring
+
+theorem wronskian_eq_of_sum_zero {a b c : R[X]} (h : a + b + c = 0) :
+    wronskian a b = wronskian b c := by
+  rw [← neg_eq_iff_add_eq_zero] at h
+  rw [← h]
+  rw [wronskian_neg_right]
+  rw [wronskian_add_right]
+  rw [wronskian_self]
+  rw [add_zero]
+  rw [← wronskian_anticomm]
```

Comment:
Can you combine these into a single rewrite chain?

Reply:
I combined all the `rw`s *except for the first one*, is it okay or could you help me to combine two further?
https://github.com/leanprover-community/mathlib4/blob/9861215de2277c9d3341cd47e5aa9c3ac0dbc880/Mathlib/RingTheory/Polynomial/Wronskian.lean#L59-L60